File: Java/FizzBuzz.java
import java.util.Arrays;
import java.util.Scanner;

/**
 * Created by August DH on 2016-12-11.
 */
public class FizzBuzz {
    public static void main(String[]args){
        //Input
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();

        //Splitting
        String[] splittedString = input.split(" ");

        //Parsing
        int[] intArray = {Integer.parseInt(splittedString[0]),
                          Integer.parseInt(splittedString[1]),
                          Integer.parseInt(splittedString[2])
        };

        //System.out.println(Arrays.toString(intArray));

        //Fills up the interwall
        int[] interwall = new int[intArray[intArray.length-1]];
        for (int i = 0; i < intArray[intArray.length-1]; i++){
                interwall[i] = i +1;
        }

        //System.out.println(Arrays.toString(interwall));

        //Calculates which numbers are dividable with the numbers in the interwall
        String[] returnArrayString = new String[interwall.length];
        for(int i = 0; i<interwall.length; i++){
            returnArrayString[i] = "" +(i+1);
            if((i+1) % intArray[0] == 0){
                returnArrayString[i] = "Fizz";
            }
            else if ((i+1) % intArray[1] == 0){
                returnArrayString[i] = "Buzz";
            }
            if((i+1)% intArray[0] == 0 && (i+1) % intArray[1] == 0){
                returnArrayString[i] = "FizzBuzz";
            }

        }

      //  System.out.println(Arrays.toString(returnArrayString));

        String returnString = "";
        for(int i = 0; i<returnArrayString.length; i++){
            System.out.println(returnArrayString[i]);
        }

    }
}

File: Python/fizzbuzz.py
N,M = list(map(int, input().split()))

correct_answers = []
for i in range(1,M+1,1):
    if(i % 3 == 0 and i % 5 == 0):
        correct_answers.append("fizzbuzz")
    elif(i%3 == 0):
        correct_answers.append("fizz")
    elif(i%5 == 0):
        correct_answers.append("buzz")
    else:
        correct_answers.append(str(i))

most_correct = 0
best_cand = 1

for candidate in range(N):
    candidate_string = input().split()
    points = 0

    for i in range(len(candidate_string)):
        if(candidate_string[i] == correct_answers[i]):
            points += 1

    if(points > most_correct):
        best_cand = candidate+1
        most_correct = points

print(best_cand)
